src/Talis/Persona/Client/Tokens.php

```diff
@@ -264,23 +271,99 @@ public function getUserByGuids($guids, $token)
         {
             throw new \InvalidArgumentException("Invalid guids");
         }
-        if(trim($token) === '')
+        if(!is_string($token) || trim($token) === '')
         {
             throw new \InvalidArgumentException("Invalid token");
         }
         $url = $this->config['persona_host'].'/users/?guids='.implode(',', $guids);
```

Comment:
probably not one for this story, but there will be a point where this URL gets too long, dependent on the number of guids passed to fetch